cpp/ps0004/ps0004_soin_cpp.cpp

```diff
@@ -2,17 +2,27 @@
 #include <conio.h>
 using namespace std;
 int main() {
-	int num;
+	// prompting user for number 1/1.5
+	float num;
 	cout << "Enter a number (ASCII value between 32 and 126) \n";
 	cin >> num;
 
-	if (num >= 32 && num <= 126) {
+	// checking invalidaty of user's input 1.5/4
+	// printing the number and the opposite printable character 1.5/2
+	if(!num) {
+		cout << "Invalid character/s or zero value!\n";
+	}
+	else if((int) num != num) {
+		cout << "Invalid: a floating point number!\n";
+	}
+	else if(num >= 32 && num <= 126) {
 		char ch = static_cast<char>(num);
-		cout << "Equivalent character \n" << ch << endl;
+		cout << "Equivalent character for the number " << num << " is " << ch << "." << endl;
 	}
 	else {
-		cout << "Invalid input. ASCII value must be between 32 and 126\n" << std::endl;
+		cout << "Invalid: number is out of the range!\n";
 	}

```

Comment:
- [x] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [ ] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [ ] printed the number and the opposite printable character